src/functionals.jl

```diff
@@ -13,9 +13,17 @@ end
 
 ## Assuming that discrete distributions only take integer values.
 function expectation(distr::DiscreteUnivariateDistribution, g::Function, epsilon::Real)
-    f = x->pdf(distr,x)
     (leftEnd, rightEnd) = getEndpoints(distr, epsilon)
-    sum(x -> f(x)*g(x), leftEnd:rightEnd)
+    sum(leftEnd:rightEnd) do x
+        pdf(distr,x) * g(x)
+    end
+end
+
+function expectation(distr::CountableUnivariateDistribution,
+                     g::Function, epsilon::Real)
+    sum(support(distr)) do x
```

Comment:
Does this make sense, this is a loop over typically 2ˆ64 elements?

Reply:
This is just a slight editing of the original code:
```julia
    f = x->pdf(distr,x)
    (leftEnd, rightEnd) = getEndpoints(distr, epsilon)
    sum(x -> f(x)*g(x), leftEnd:rightEnd)
```
I don't have a strong opinion about whether it's sensible.

Reply:
You are dropping the epsilon

Reply:
Ah, I see. That's because this is for non-`ContiguousSupport` types - they are not necessarily ordered, so we can't use `getEndpoints()`. At the moment, the only type that falls into this category is `DiscreteNonParametric`, but it will also include `Dirac`, neither of which are expected to have anything like 2^64 elements..